opentelemetry-otlp/src/exporter/http/metrics.rs

```diff
@@ -48,20 +48,28 @@ impl MetricsClient for OtlpHttpClient {
     }
 }
 
-#[cfg(feature = "http-proto")]
+#[cfg(any(feature = "http-proto", feature = "http-json"))]
 fn build_body(metrics: &mut ResourceMetrics) -> Result<(Vec<u8>, &'static str)> {
```

Comment:
For this part. I wonder if we can make the feature additive. 

One way I see we can do it is:

- Add `protocol` field in `OtelHttpClient`
- Move `build_body` as part of the `OtelHttpClient`.

Then we just ask users to pass protocol in building pipline(it's already part of the `export_config`. And based on the feature the corresponding protocol may be available

Reply:
Are we still talking about exporting in one format per signal per pipeline? If so, it should be possible with the above code, no? For e.g I can enable both features `http-json` and `http-photo` and build 2 trace pipelines that export in each of those formats. Let me know if I'm missing something.

Reply:
Yes by tuning the features it's possible to export in either `proto` or `json`. My preference is to ask users to explicitly tell the pipeline which protocol to rather then to config it using features. i.e it should be possible to enable both `http-json` and `http-proto` but choose to use `http-proto`. I think it current setup doesn't support this.

Reply:
Understood. Would you want `HttpExporterBuilder` to receive protocol via `with_protocol`?